### ISSUE TITLE
Sparsity_sc and sparsity_sp = 0

### ISSUE THREAD

**Issue by gboscagli** [broadinstitute/Tangram#118]
Greetings,

I'm using Tangram to map a scRNA-seq dataset of mouse lung into a spatial dataset (Visium 10X) of the same specimen. Both datasets are preprocessed using standard Scanpy pipelines. I followed the [Tangram tutorial available in the Squidpy docs](https://squidpy.readthedocs.io/en/stable/notebooks/tutorials/tutorial_tangram.html), so I segmented the cells using watershed and used the resulting cell count to tailor the `target_count` and `density_prior` for `map_cells_to_space`, as shown in the linked tutorial, with `mode="constrained"`. The training genes are 1K, as expected. Once I visualize the diagnosis plots using `plot_training_scores`, however, the result is the following.

![image](https://github.com/broadinstitute/Tangram/assets/133790786/a1928639-c5f5-4093-b9df-c2b5d8b6b0a5)

Basically, all the training scores are between 0.952 and 0.999 because `sparsity_sc` and `sparsity_sp` are both 0 for all genes, and the predictions on test genes yield an `auc_score` of 0. This issue happens for other two pairs of scRNA-seq/spatial datasets that I'm using, so I think I'm missing some steps in the pipeline.

I'm wondering which could be the reason for such behaviour. Hope you can help me, I would really appreciate it!

**Comment by Yunxia-wang** [comment]
I have the same problem. I use the MERFISH data and scRNA-seq data from scanpy processed. I get the sparsity_sp=0 in all genes. 

`ad_map = tg.map_cells_to_space(adata_sc, adata_st,
    mode="cells",
    density_prior='uniform',
    num_epochs=1000,
    # device="cuda:0",
    device='cpu',
)`

<img width="1185" height="290" alt="Image" src="https://github.com/user-attachments/assets/7a020e92-4c71-4f56-b789-4ab90fce6c00" />

And I get error when I want to plot the auc. 
`lib/polynomial.py:638, in polyfit(x, y, deg, rcond, full, w, cov)
    636     raise TypeError("expected 1D vector for x")
    637 if x.size == 0:
--> 638     raise TypeError("expected non-empty vector for x")
    639 if y.ndim < 1 or y.ndim > 2:
    640     raise TypeError("expected 1D or 2D array for y")

TypeError: expected non-empty vector for x`
